src/data-table/data-table.js

```diff
@@ -76,6 +76,10 @@
           row.classList.add(this.CssClasses_.IS_SELECTED);
         } else {
           row.classList.remove(this.CssClasses_.IS_SELECTED);
+          var thead_checkbox = this.element_.tHead.querySelector('.mdl-checkbox.is-checked');
```

Comment:
This should be cached to avoid looking it up all the time. The selection should be done in the init method (or just cache the created node as a property).